proto/api_key.proto

```diff
@@ -0,0 +1,80 @@
+syntax = "proto3";
+
+import "proto/context.proto";
+
+package api_key;
+
+// An API key used to access BuildBuddy.
+message ApiKey {
+  // The unique ID of this API key.
+  // ex: "AK123456789"
+  string id = 1;
+
+  // The string value of this API key which is passed in API requests.
+  string value = 2;
+
+  // Optional. The user-specified label of this API key that helps them
+  // remember what it's for.
+  string label = 3;
+}
+
+message CreateApiKeyRequest {
+  context.RequestContext request_context = 1;
+
+  // The ID of the group to create the API key for.
+  // ex: "GR123456789"
+  string group_id = 2;
+}
+
+message CreateApiKeyResponse {
+  context.ResponseContext response_context = 1;
+
+  // The ID of the newly created key.
+  string id = 2;
+
+  // The value of the newly created key.
+  string value = 3;
+}
+
+message GetApiKeysRequest {
+  context.RequestContext request_context = 1;
+
+  // The ID of the group to get API keys for.
+  // ex: "GR123456789"
+  string group_id = 2;
+}
+
+message GetApiKeysResponse {
+  context.ResponseContext response_context = 1;
+
+  // The API keys owned by the requested group.
+  repeated ApiKey api_key = 2;
+}
+
+message UpdateApiKeyRequest {
```

Comment:
Let's say that in the future we add some other attributes to API Keys, like "permissions" which could be read | write.

How would a client request the server to change an API key's permissions bit, without having to set the "label" field, which they do not want to change?

Reply:
Ideally we want to be able to provide a list of `Update` objects which are `(field, newValue)` tuples, but it seems a bit overkill to implement that ourselves -- there might be frameworks out there that make it easier to do this in a type-safe way (e.g. provide field references instead of string constants). We had an internal framework that did this at one of the teams I was working on at Google and it worked pretty well.

FWIW we have the same problem with the `Groups` proto -- in the `UpdateGroupRequest` we pass in all the values of all the fields. We figured it's not a big deal since this is a low-throughput RPC. Seems like that'd be the case with the API key RPCs as well? I'd prefer sticking with the current approach since it's less work and easier for the client to send all the values in each request anyway.

Reply:
OK, I'm OK with leaving this as-is for the reasons you stated (low volume, simplicity).

In more "real" code, we should be using a field mask which is the proto-3 way to handle this. Or just using proto2 lol which doesn't have the limitations of proto3.